atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/service/TransactionServices.java

```diff
@@ -95,6 +112,16 @@ private static TransactionService createV3TransactionService(
                 SimpleTransactionService.createV3(keyValueService, metricRegistry, acceptStagingReadsAsCommitted)));
     }
 
+    private static TransactionService createV4TransactionService(
+            KeyValueService keyValueService,
+            TaggedMetricRegistry metricRegistry,
+            Supplier<Boolean> acceptStagingReadsAsCommitted,
+            Optional<InternalSchemaInstallConfig> config) {
+        return new PreStartHandlingTransactionService(
+                WriteBatchingTransactionService.create(SimpleTransactionService.createV4(
```

Comment:
So we don't use detail on failure in v4 anyway so might as well just not use write batching and use the MCAS batching